### DISCUSSION TITLE
[Inquire] .cake vs .targets

### DISCUSSION THREAD

**Discussion by WanftMoon** [AdamEssenmacher/GoogleApisForiOSComponents#12] [Q&A] [answered]
Hi @AdamEssenmacher,

First of all, tyvm for your work!!

I've an inquire related to how the .xcframeworks are being included. Was trying to understand how the deps are figured out when things get build, and noticed that we have:

1. [components.cake](https://github.com/AdamEssenmacher/GoogleApisForiOSComponents/blob/main/components.cake)
that uses cocoapods to download the xcframeworks.

but then there is also:

2. [*.targets](https://github.com/AdamEssenmacher/GoogleApisForiOSComponents/blob/main/source/Firebase/Analytics/Analytics.targets) for each project (in this case, analytics)
that also downloads stuff from an URL

and I also noticed that the cake is v19 while targets is v17. Who is priority in the build order? is *.target being used only when building outside of cake?

**Comment by AdamEssenmacher** [comment, marked as answer]
The .cake toolchain is used to download most .xcframeworks, build the .NET binding libraries, and pack the nugets.

The .target files are packed into the nugets to be consumed by MSBuild in dependent projects (e.g. 'YourMauiApp.csproj'). The .targets files tell MSBuild where to find the native libraries (the xcframeworks) and which framework/linker flags to use.

Most xcframeworks are packed directly into the nugets as resources. However, in some cases (notably in Firebase, Analytics, and GoogleAppMeasurement), the xcframeworks are downloaded directly from Google during the build. This is because these are shipped as `vendored_frameworks` (i.e. prebuilt binaries), as you can see in the [FirebaseAnalytics podspec](https://github.com/CocoaPods/Specs/blob/master/Specs/e/2/1/FirebaseAnalytics/10.19.0/FirebaseAnalytics.podspec.json)

You can also note that the 10.19 podspec linked above references the 10.17 binary. This is because FirebaseAnalytics hasn't changed since 10.17, so the source binary is re-used.

**Comment by WanftMoon** [comment]
Ok, now i got it. Analytics only changed the binaries on v20. thought they would had been 1 to 1, but in this case they kept it up to v19. Damn, this stuff is nuts. too much details and each package may have its changes or not.
![Screenshot 2024-04-27 at 14 34 20](https://github.com/AdamEssenmacher/GoogleApisForiOSComponents/assets/20956737/2d8cd3cb-804b-4410-965b-fd16b8d5cfe7)
